cmake/host-tools.cmake

```diff
@@ -1,5 +1,17 @@
 include(${ZEPHYR_BASE}/cmake/host-tools-zephyr.cmake)
 
+# west is optional
+find_program(
+  WEST
+  west
+  )
+if(${WEST} STREQUAL WEST-NOTFOUND)
+  unset(WEST)
+  message(STATUS "Unable to find west, some functionality may not be available. \
```

Comment:
This is going to be very annoying (cmake is already way too verbose). Please remove this.
Won't the user get a clear error if he tries to run `ninja flash` and west is not found?

Reply:
Yeah, I think this is another thing that could be solved by moving the functionality into the Zephyr repository.

West in turn calls back to CMake as far as I can tell. That makes it very twisty and hard to follow.

Reply:
Fair enough points from both. I will remove it.

Reply:
> This is going to be very annoying (cmake is already way too verbose). Please remove this.
> Won't the user get a clear error if he tries to run `ninja flash` and west is not found?

CMake now detects whether `west` is available and useable and issue a corresponding message when issueing, e.g. `ninja flash/debug/...`.

Example:
```
West was not found in path. To support '/usr/bin/ninja flash', please install west bootstrapper with: 'pip install west --user', and thereafter 'west init'.
```